dig.go

```diff
@@ -128,6 +129,26 @@ func Group(group string) ProvideOption {
 	})
 }
 
+// ConstructorInfo provides information about the constructor's inputs and outputs
+// types as strings, as well as the ID of the constructor supplied to the Container.
+//
+// ID is a unique integer representing the constructor node in the dependency graph.
+// Inputs is a string slice with the types of the input parameters of the constructor.
+// Outputs is a string slice with the types of the results produced by the constructor.
+type ConstructorInfo struct {
+	ID      int
+	Inputs  []string
+	Outputs []string
+}
+
+// WithInfo is a ProvideOption that writes info on what Dig was able to get out
+// out of the provided constructor into the provided ConstructorInfo.
+func WithInfo(info *ConstructorInfo) ProvideOption {
```

Comment:
Couple things: The other ProvideOptions don't have a With prefix. The divergence is fine because those were written before we agreed that the prefix was desirable on most options, but in this case, "WithFoo" usually suggests you're supplying Foo, not requesting Foo. Should we name this something else? Like, `FillConstructorInfo` (or better?—that's just the first idea that came to mind).

Reply:
That's fair. I'll go with `FillInfo`.